docs/dev/srs.md

```diff
@@ -129,63 +94,100 @@ in several areas:
+This allows flexibility to translate DFN specifications to a different format, e.g. TOML, YAML or JSON.
 
-Component classes also provide a data access layer via `.get_data()` and
-`set_data()` &mdash; it would be simpler just to get/set the attributes
-normally and intercept these behind the scenes for any magic necessary.
+Component classes also provide a data access layer via `.get_data()` and `set_data()` &mdash; it would be simpler just to get/set the attributes normally and intercept these behind the scenes for any magic necessary.
 
 ### Performance
 
 TODO: describe current issues
 
 ### Invariants
 
-FloPy 3 has a "check" mechanism for validating simulation configurations,
-but this must be run manually by the user, and no straightforward method
-for extension is available.
+FloPy 3 has a "check" mechanism for validating simulation configurations, but this must be run manually by the user, and no straightforward method for extension is available.
+This allows simulations to be initialized in an invalid state, which may go unnoticed until runtime, producing less than informative errors from the modeling program.
+The most obvious example is that grid dimensions can be changed with no warning to the user, and no attempt to coerce package array data to the new shape.
+We would like automatic enforcement of invariants whenever a simulation changes.
 
-This allows simulations to be initialized in an invalid state, which may
-go unnoticed until runtime, producing less than informative errors from
-the modeling program.
+### Maintainability
 
-The most obvious example is that grid dimensions can be changed with no warning to the user, and no attempt to coerce package array data to the
-new shape.
-
-We would like automatic enforcement of invariants whenever a simulation
-changes.
+flopy4 should maintain a separation of concerns with respect to hydrology and software engineering.
+The hydrologic modeler should not be concerned with the technical details of flopy4's internal data storage or parallel processing implementation,
+for example, and a software engineer should be able to work on the code without detailed knowledge on complex hydrologic concepts.
 
 ## Goals
 
-With the above in mind, we want the next version of FloPy to
+With the above in mind, we want the next version of FloPy to:
 
 - preserve existing `flopy.mf6` functionality
-- be consistent, user-friendly and Pythonic
+- be consistent, user-friendly and pythonic
 - be easy to read, debug, diagnose and test
 - be memory-efficient and provide fast IO
 - impose a minimal maintenance burden
+
+## System requirements and functional requirements
+
+### Functional requirements
+
+| ID     | Description | MoSCoW |
+| ------ | ----------- | ------ |
+| FUNC-1 | flopy4 must be able to read and write MODFLOW 6 input files and read MODFLOW 6 output files. | M |
```

Comment:
Maybe we need another one to say: The results from a model that is loaded by flopy and written back out must give the same results as the original model.

Reply:
That's what I was thinking, @wpbonelli.  We've struggled over the years with loading a model, writing it back out, and getting the same results as before (to within some level of precision).

Reply:
Changed to:

> Flopy is a non-intrusive package when it comes to reading, writing, and running the model. When writing out a model that was read, the simulation output must give the exact same results.